### ISSUE TITLE
`FlaxStableDiffusionPipeline` crashes when trying to load SDXL

### ISSUE THREAD

**Issue by davisyoshida** [huggingface/diffusers#4303]
### Describe the bug

When loading "stabilityai/stable-diffusion-xl-base-1.0" with `from_pretrained`, I get the following error:
```
Error no file named diffusion_flax_model.msgpack or diffusion_pytorch_model.bin found in directory /root/.cache/huggingface/hub/models--stabilityai--stable-diffusion-xl-base-1.0/snapshots/eef8a746a4f8cce8e557ffa34cac849205ba6e4f/vae.
```

It looks like the files are in the safetensors format, so assuming the error message it looks like maybe this pipeline doesn't check for that?

### Reproduction

```python
pipeline, params = FlaxStableDiffusionPipeline.from_pretrained(
    'stabilityai/stable-diffusion-xl-base-1.0',
)
```

### System Info

- `diffusers` version: 0.19.0
- Platform: Linux-5.15.109+-x86_64-with-glibc2.35
- Python version: 3.10.6
- PyTorch version (GPU?): 2.0.1+cu118 (True)
- Huggingface_hub version: 0.16.4
- Transformers version: 4.31.0
- Accelerate version: not installed
- xFormers version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@pcu

**Comment by patrickvonplaten** [comment]
cc @pcuenca

**Comment by rajbala** [comment]
I get the same error with the StableDiffusionPipeline:  `pipe = StableDiffusionPipeline.from_pretrained(model, cache_dir="/app/hf_hub_cache", torch_dtype=torch.float16).to("cuda")`

`OSError: Error no file named diffusion_pytorch_model.bin found in directory /app/hf_hub_cache/models--stabilityai--stable-diffusion-xl-base-1.0/snapshots/47cd5302d866fa60cf8fb81f0e34d42e38f6100c/vae.`

**Comment by Johanan528** [comment]
> I get the same error with the StableDiffusionPipeline: `pipe = StableDiffusionPipeline.from_pretrained(model, cache_dir="/app/hf_hub_cache", torch_dtype=torch.float16).to("cuda")`
> 
> `OSError: Error no file named diffusion_pytorch_model.bin found in directory /app/hf_hub_cache/models--stabilityai--stable-diffusion-xl-base-1.0/snapshots/47cd5302d866fa60cf8fb81f0e34d42e38f6100c/vae.`

Have you solved this bug?

**Comment by pcuenca** [comment]
Hello @Johanan528, @rajbala!

To load the PyTorch SDXL pipeline, you can use `StableDiffusionXLPipeline`, `DiffusionPipeline`, or `AutoPipelineForText2Image`, but not `StableDiffusionPipeline` 😅 The latter is appropriate for any of the classic Stable Diffusion models, including versions `2` and `2.1`, but SDXL is different and requires its own pipeline. You can read more about how to load and use SDXL in [the latest version of our documentation](https://huggingface.co/docs/diffusers/main/en/using-diffusers/sdxl).

Regarding Flax, SDXL is still being ported, so you can't load a Flax pipeline yet.  If you need it, you may take a look at [the development branch](https://github.com/huggingface/diffusers/pull/4254) and install from source :)

**Comment by ngaer** [comment]
Hey @pcuenca, thanks for the clarification! Do you have any eta on when this Flax pipeline for SDXL will be done?

**Comment by pcuenca** [comment]
Fixed in #4254
